pkg/thanos/statefulset.go

```diff
@@ -294,7 +296,26 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 	if tr.Spec.AlertQueryURL != "" {
 		trCLIArgs = append(trCLIArgs, monitoringv1.Argument{Name: "alert.query-url", Value: tr.Spec.AlertQueryURL})
 	}
-
+	if len(tr.Spec.Version) > 0 {
+		version, err := semver.ParseTolerant(tr.Spec.Version)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse thanos ruler version")
+		}
+		if version.GTE(semver.MustParse("0.24.0")) {
+			if tr.Spec.RemoteWrite != nil {
+				for i, remote := range tr.Spec.RemoteWrite {
+					if err := prometheus.ValidateRemoteWriteSpec(remote); err != nil {
+						return nil, errors.Wrapf(err, "validate remote write %d", i)
+					}
+				}
+				remoteWriteYaml, err := yaml.Marshal(generateRemoteWriteConfigYaml(tr.Spec.RemoteWrite))
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to generate remote write spec")
+				}
+				trCLIArgs = append(trCLIArgs, monitoringv1.Argument{Name: "remote-write.config", Value: string(remoteWriteYaml)})
```

Comment:
On `pkg/prometheus` we already have a method for generating RemoteWrite config YAML. Maybe we could re-use it?

Reply:
I think configGenerator.generateRemoteWriteConfig does not work with thanos ruler.

Reply:
Could you elaborate on why you think so? I could be wrong, but so far I understood that they are equal (they even have the same spec)